### PR TITLE
Fix for 17509. Do not allow migration to have same name as context.

### PR DESCRIPTION
**Pull request by lajones**
@bricelam - this is a quick fix for #17509 - to error if you try to add a migration with the same name as the context, and ask the user to pick a different migration name. Or did you want the more complicated solution of trying to write out code that adds namespaces to correctly resolve the reference in the generated code?

### CODE REVIEW

**Comment by bricelam** [comment]
Hmm, between not generating code (this) and generating code that doesn't compile but I can easily fix (the current behavior), I think I'd rather have bad code.

How complicated would it be to scaffold `[DbContext(typeof(The.Full.Namespace.Of.TheDbContext))]` as needed instead?

**Comment by lajones** [comment]
Well, the user can easily fix it under this approach (i.e. throw the exception) by re-running Add-Migration with a different name. But to address your comments. If you want to output the full namespace on every migration it's probably easy. If only when there's a chance of a clash - I imagine a bit harder but not impossible.

**Comment by lajones** [comment]
See also #20132 for the approach where we only put the namespace of the `DbContext` class in the `DbContext` attribute if it clashes with the migration name.

File: src/EFCore.Design/Properties/DesignStrings.resx [hunk 1]
```diff
@@ -363,4 +363,7 @@ Change your target project to the migrations project by using the Package Manage
   <data name="UsingHostingServices" xml:space="preserve">
     <value>Using application service provider from Microsoft.Extensions.Hosting.</value>
   </data>
+  <data name="ConflictingContextAndMigrationName" xml:space="preserve">
+    <value>The name you have chosen for the migration, '{name}', is the same as the context class name. Please choose a different name for your migration.</value>
```
Comment (bricelam) [inline src/EFCore.Design/Properties/DesignStrings.resx]: lol, should we add something like this? "We suggest using 'InitialCreate' for the first migration."

Comment (lajones) [reply src/EFCore.Design/Properties/DesignStrings.resx]: That's fine by me (unless of course that's what they've called their context class 😄).

**Comment by lajones** [comment]
@bricelam Can you sign-off on #20143 instead please. It's the same fix - this branch just got corrupted due to merge conflicts and now it won't let me re-open this PR.

**Pull request closed without merging**